Sources/Purchasing/Purchases/PurchasesOrchestrator.swift

```diff
@@ -998,17 +997,14 @@ private extension PurchasesOrchestrator {
             self.operationDispatcher.dispatchOnWorkerThread {
                 let hasTransactions = self.transactionsManager.customerHasTransactions(receiptData: receiptData)
                 let cachedCustomerInfo = self.customerInfoManager.cachedCustomerInfo(appUserID: currentAppUserID)
-                let hasOriginalPurchaseDate = cachedCustomerInfo?.originalPurchaseDate != nil
 
-                if !hasTransactions && hasOriginalPurchaseDate {
+                if !hasTransactions, let customerInfo = cachedCustomerInfo, customerInfo.originalPurchaseDate != nil {
                     if let completion = completion {
                         self.operationDispatcher.dispatchOnMainThread {
-                            completion(
-                                Result(cachedCustomerInfo,
-                                       ErrorUtils.customerInfoError(withMessage: "No cached customer info"))
-                            )
+                            completion(.success(customerInfo))
```

Comment:
@aboedo this new implementation is 100% equivalent. You can see that it wasn't even possible to send an error in this particular case.